stubs/docutils/docutils/transforms/__init__.pyi

```diff
@@ -1,14 +1,31 @@
-from _typeshed import Incomplete
+from collections.abc import Iterable
+from types import ModuleType
+from typing import Any
 
-from docutils.nodes import Node, document
+from docutils import ApplicationError, Component, TransformSpec, nodes
+from docutils.nodes import Node
+
+class TransformError(ApplicationError): ...
 
 class Transform:
-    def __init__(self, document: document, startnode: Node | None = None): ...
-    def __getattr__(self, name: str, /) -> Incomplete: ...
+    default_priority: int | None
+    document: nodes.document
+    startnode: Node | None
+    language: ModuleType
+    def __init__(self, document: nodes.document, startnode: Node | None = None) -> None: ...
+    def apply(self, **kwargs: Any) -> None: ...
 
-class Transformer:
-    def __init__(self, document: document): ...
+class Transformer(TransformSpec):
+    transforms: list[tuple[str, type[Transform], Node | None]]
+    document: nodes.document
+    applied: list[tuple[int, type[Transform], nodes.pending, Any]]
+    sorted: bool
+    components: dict[str, Component]
+    serialno: int
+    def __init__(self, document: nodes.document) -> None: ...
     def add_transform(self, transform_class: type[Transform], priority: int | None = None, **kwargs) -> None: ...
-    def __getattr__(self, name: str, /) -> Incomplete: ...
-
-def __getattr__(name: str) -> Incomplete: ...
+    def add_transforms(self, transform_list: Iterable[type[Transform]]) -> None: ...
+    def add_pending(self, pending: nodes.pending, priority: int | None = None) -> None: ...
+    def get_priority_string(self, priority: int) -> str: ...
```

Comment:
At runtime, you *should* not have priority = None but with the implementation you could have it. The reason is that `default_priority` of a transformation class must be overridden in subclasses but it is *not* enforced.

You somehow enforce the fact that the transformation class lacks an explicit `default_priority` value as follows:

- Use `add_trasnform(t)` without *priority*.
- The implementation uses ``p = t.default_priority``. It is `None` but directly calls `get_priority_string(p)`.
- Format ``'%03d-%03d' % (priority, self.serialno)``. Since None is not '%d'-valid, you get an exception. 

Forcing `int` as the argument type would make docutils fail on that. However, using `int | None` to make it pass could make people assume that a None priority is still allowed. Would it make sense to have something like:

```suggestion
    @overload
    def get_priority_string(self, priority: None) -> NoReturn: ...
    @overload
    def get_priority_string(self, priority: int) -> str: ...
```
?

Reply:
i wish we could just refactor docutils...

Reply:
That's unfortunately a separate issue. Maybe a -1 would have been a better default priority.

Reply:
could we mark it as an `@abstractproperty` and then suppress the stubcheck error?

```python
    @property
    @classmethod
    @abstractmethod
    def default_priority(cls) -> int: ...
```

trouble is, a get a lint warning that `classmethod` is deprecated in python 3.11 and removed in python 3.13. not sure what the alternative is?